### PR TITLE
Feature/null safety

### PR DESCRIPTION
**Pull request by Dave0921**
Migrate to null safety.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.0.6
```
Comment (wolfenrain) [inline CHANGELOG.md]: As the null-safety support is a breaking change I think it should increase a minor patch:
```suggestion
## 0.1.0
```

File: pubspec.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name: ocarina
 description: Play local (assets or external/internal storage) audio files with flutter on Android or iOS
-version: 0.0.5
+version: 0.0.6
 homepage: https://github.com/erickzanardo/ocarina
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.12.0 <3.0.0'
```
Comment (wolfenrain) [inline pubspec.yaml]:
```suggestion
  sdk: ">=2.12.0 <3.0.0"
```

File: lib/ocarina.dart [hunk 3]
```diff
@@ -68,7 +68,7 @@ class OcarinaPlayer {
         'seek', {'playerId': _id, 'position': duration.inMilliseconds});
   }
 
-  Future<int> position() async {
+  Future<int?> position() async {
```
Comment (wolfenrain) [inline lib/ocarina.dart]: The channel method will always return a int so the type here should reflect that, and the return value should be unpacked.

**Review by wolfenrain — CHANGES_REQUESTED** [review]
Few comments. I think it is also important to migrate the example to ensure everything is working as intended.

**Comment by Dave0921** [comment]
> Few comments. I think it is also important to migrate the example to ensure everything is working as intended.

Good point. I'll make sure to do that.

File: lib/ocarina.dart (hunk 3 above)
Comment (Dave0921) [reply lib/ocarina.dart]: In the native layer there is a possibility of returning null; for instance under the guard clause here: https://github.com/erickzanardo/ocarina/blob/3b5e668dce959d481ae445492cc9301534080140/ios/Classes/SwiftOcarinaPlugin.swift#L418

I can unpack it in this method and raise an exception (`PlatformException`) if the int result returned from the native layer is null. Would that be the best approach for this case?

Comment (erickzanardo) [reply lib/ocarina.dart]: I agree on raising an exception. It should never happen for an id be passed to the native layer, and the player don't exist there.

File: lib/ocarina.dart [hunk 4]
```diff
@@ -70,7 +70,12 @@ class OcarinaPlayer {
 
   Future<int> position() async {
     _ensureLoaded();
-    return await _channel.invokeMethod('position', {'playerId': _id});
+    final pos = await _channel.invokeMethod('position', {'playerId': _id});
+    if (pos == null) {
+      throw PlatformException(
+          code: 'PositionNull', message: 'Position is null');
```
Comment (wolfenrain) [inline lib/ocarina.dart]: Missing trailing comma :-)

**Review by wolfenrain — APPROVED** [review]
LGTM

**Review by erickzanardo — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**